feat: add page count component

NONONOexe opened the pull request:
To help maintain note-taking motivation, knowing the total number of pages is important. This commit introduces a new `PageCount` component that displays the total number of pages (notes).

Translations have been added for each locale, referencing existing translations as a guide. However, being a native Japanese speaker, I may not have accurately translated the terms for locales other than English and Japanese. Please review the translations for other languages.

NONONOexe committed:
feat: add page count component

新しいPageCountコンポーネントを追加し、各言語のローカライズファイルにページ数の翻訳を追加しました。

NONONOexe committed:
fix: format PageCount with Prettier

saberzero1 commented:
Hey, several languages have options to distinguish singular and plural. (E.g. `1 note`/`2 notes`) You seem to remove these options as well as change unrelated translations.

NONONOexe committed:
:fix: Undo accidental contentMeta modification

NONONOexe committed:
Merge branch 'page-count' of https://github.com/NONONOexe/quartz into page-count

NONONOexe committed:
Merge branch 'page-count' of https://github.com/NONONOexe/quartz into page-count

NONONOexe committed:
Merge branch 'page-count' of https://github.com/NONONOexe/quartz into page-count

NONONOexe commented:
Hi @saberzero1,

Thanks for the feedback. I've made the following updates:

- Reverted the accidental modification to `contentMeta` (f6d990f57a3558a74461f73908e6ed454b8d7230).
- Adjusted the translation content to align with the style of existing translations (1305a64ba6e3df2bc452dfca06967a33d5779c4a).

jackyzha0 commented:
thanks for the contribution but i am closing this as I think this is a relatively simple component for downstream users to implement and adds some internationalization burden that is outsized for its benefit

i do think this serves as a good example for future component implementers though so thank you for that!

NONONOexe commented:
Thank you for your reply and feetback! I understand and agree with your assessment.
